docs/src/pages/en/styling-reference.astro

```diff
@@ -0,0 +1,107 @@
+---
+import { Markdown } from '@astropub/md';
+import MainLayout from '../../layouts/MainLayout.astro';
+import manifest from '../../../../dist/custom-elements.json';
+
+const { frontmatter = {}, headings = [], url } = Astro.props as Props;
+const components = await Astro.glob('./components/*');
+
+function hasLink(tagName) {
+  return components.some(c => c.file.includes(tagName));
+}
+
+const groups = [
+  'button',
+  'range',
+  'display',
+  'indicator',
+  'other',
+];
+
+for (let group of groups) {
+  headings.push({ depth: 2, slug: `${group}s`, text: `${capitalizeFirstLetter(group)}s` });
+}
+
+let allCssProps = {};
+for (let cls of getClasses(manifest)) {
+  for (let cssProp of cls.cssProperties ?? []) {
+    if (!allCssProps[cssProp.name]) {
+      allCssProps[cssProp.name] = cssProp;
+      cssProp.groups = new Set();
+    }
+    if (cls.tagName) {
+      allCssProps[cssProp.name].groups.add(cls.tagName.split('-').pop());
+    }
+  }
+}
+
+let sortedCssProps = [...Object.keys(allCssProps)].sort();
+
+function getClasses(manifest) {
+  let classes = [];
+  for (let moduleDoc of manifest.modules) {
+    classes.push(...moduleDoc.declarations.filter(d => d.kind === 'class'));
+  }
+  return classes;
+}
+
+function capitalizeFirstLetter(str) {
+  return str[0].toUpperCase() + str.slice(1);
+}
+
+---
+
+<MainLayout frontmatter={frontmatter} headings={headings}>
+  <h1>Styling Reference</h1>
+
+  {groups.map(group => (
+    <Markdown of={`## ${capitalizeFirstLetter(group)}s`} />
+
+    <ul>
+      {getClasses(manifest)
+        .filter(cls => {
+          if (group !== 'other') {
+            return cls.tagName?.endsWith(group);
+          }
+          return cls.tagName && groups.every(group => !cls.tagName?.endsWith(group));
+        })
+        .map(cls => (
```

Comment:
suggestion(non-blocking): I think it would be nice to ensure that the "base class" components show up at the end (maybe with a simple description of "Base class" or similar?). Based on our naming convention, we can probably get away with matching against `name.startsWith('media-chrome-')`?